contract/views/res_partner_view.xml

```diff
@@ -19,7 +19,7 @@
                     groups="account.group_account_invoice"
                     icon="fa-book"
                     context="{'default_contract_type': 'sale', 'contract_type': 'sale'}"
-                    attrs="{'invisible': [('customer_rank','=',0)]}"
+                    attrs="{'invisible': [('sale_contract_count','=',0)]}"
```

Comment:
Let's see if whith this we can find a consensus solution:
```suggestion
                    attrs="{'invisible': [('customer_rank','=',0), ('sale_contract_count','=',0)]}"
```

Reply:
Ouch, I would say this is worse...